### PR TITLE
Fix EOF error on audits (storage node)

### PR DESCRIPTION
**Pull request by kaloyan-raev**
`io.CopyN` may return `io.EOF` even after reading all necessary data. The reason is that `io.CopyN` was told to copy more bytes than available.

This PR also improves the debug logs to show the namespaced piece id, so it is easier to diagnose problems on the storage node.

### CODE REVIEW

File: pkg/piecestore/psserver/retrieve.go [hunk 1]
```diff
@@ -168,18 +168,18 @@ func (s *Server) retrieveData(ctx context.Context, stream pb.PieceStoreRoutes_Re
 
 		used += nextMessageSize
 		n, err := io.CopyN(writer, storeFile, nextMessageSize)
+		if err != nil && err != io.EOF {
```
Comment (egonelbre) [inline pkg/piecestore/psserver/retrieve.go]: Doesn't this now end up in infinite loop when file returns io.EOF and we haven't reached the length?

Comment (egonelbre) [reply pkg/piecestore/psserver/retrieve.go]: I think on error we should still increase allocationTracking. I suspect it will cause problems otherwise.

Comment (kaloyan-raev) [reply pkg/piecestore/psserver/retrieve.go]: Hmm... I don't know.

I also read in the `io.CopyN` that 

> On return, written == n if and only if err == nil

So, it's seems I am fixing the wrong thing.

Comment (aleitner) [reply pkg/piecestore/psserver/retrieve.go]: Move the EOF check here to  where we check the err from `s.retrieveData`

**Comment by kaloyan-raev** [comment]
The latest commit takes another approach. Instead of checking for EOF, io.Copy is invoked with the correct number of bytes to copy. This way EOF won't be returned if everything necessary is copied.

File: pkg/piecestore/psserver/retrieve.go (hunk 1 above)
Comment (kaloyan-raev) [reply pkg/piecestore/psserver/retrieve.go]: I pushed another commit that makes sure we pass the correct number of bytes to CopyN. This way the EOF check is not necessary.

**Comment by kaloyan-raev** [comment]
@aleitner @coyle @egonelbre Please review again

**Pull request merged**